csi/moac/volume.ts

```diff
@@ -383,18 +474,16 @@ export class Volume {
     });
     // add newly found replicas to the nexus (one by one)
     const newReplicas = Object.values(this.replicas).filter((r) => {
-      return (!r.isOffline() &&
-        !childReplicaPairs.find((pair) => pair.r === r) &&
-        !this.nodeBlackList[r.pool!.node!.name]);
+      return (!r.isOffline() && !childReplicaPairs.find((pair) => pair.r === r));
     });
     for (let i = 0; i < newReplicas.length; i++) {
       try {
-        await this.nexus.addReplica(newReplicas[i]);
-        return;
+        childReplicaPairs.push({
```

Comment:
looks like we now add all replicas at once, does this cause multiple concurrent rebuilds?

Reply:
it's up to mayastor to serialize rebuilds if it needs to do that. And indeed it does that. In the future it could do rebuilds in parallel. There has been nothing to enforce just one rebuild at a time even before. Something could have triggered fsa again resulting in a new addChild call. So I don't think the situation is worse than it was in any regard.

Reply:
hmm AFAIK mayastor does not serialize rebuilds. I remember from a while back that MOAC would not add a child to the nexus if the nexus was degraded, so this must have changed before. Multiple rebuilds should work but I do remember that we did not make those tests extensive because we were told that MOAC would not do such thing. If this is now possible then we might need to add more rebuild tests to validate this.

Reply:
It was probably a misunderstanding then. MOAC would not delete a child while a rebuild is in progress. Though AFAIK there has not been such rule for adding child. I have created the ticket CAS-890 for missing tests 👍  thanks for pointing it out.

Reply:
Ah, I see, then definitely a misunderstanding!
We used to have a rebuild multiple on mayastor but it's been removed, would be good to bring that one back, I'll update that ticket to mention it.